src/otx/core/data/dataset/segmentation.py

```diff
@@ -5,22 +5,63 @@
 
 from __future__ import annotations
 
-from typing import Callable
+from typing import TYPE_CHECKING, Callable
 
 import numpy as np
 import torch
 from datumaro.components.annotation import Image, Mask
 from torchvision import tv_tensors
 
+from otx.core.data.dataset.base import Transforms
 from otx.core.data.entity.base import ImageInfo
 from otx.core.data.entity.segmentation import SegBatchDataEntity, SegDataEntity
+from otx.core.data.mem_cache import NULL_MEM_CACHE_HANDLER, MemCacheHandlerBase
+from otx.core.types.image import ImageColorChannel
 
 from .base import OTXDataset
 
+if TYPE_CHECKING:
+    from datumaro import DatasetSubset
+
 
 class OTXSegmentationDataset(OTXDataset[SegDataEntity]):
     """OTXDataset class for segmentation task."""
 
+    def __init__(
+        self,
+        dm_subset: DatasetSubset,
+        transforms: Transforms,
+        mem_cache_handler: MemCacheHandlerBase = NULL_MEM_CACHE_HANDLER,
+        mem_cache_img_max_size: tuple[int, int] | None = None,
+        max_refetch: int = 1000,
+        image_color_channel: ImageColorChannel = ImageColorChannel.RGB,
+        stack_images: bool = True,
+    ) -> None:
+        super().__init__(
+            dm_subset,
+            transforms,
+            mem_cache_handler,
+            mem_cache_img_max_size,
+            max_refetch,
+            image_color_channel,
+            stack_images,
+        )
+        # Check & Update background label in LabelInfo
+        self._update_meta_info()
```

Comment:
Add a dedicated `meta_info` class for the semantic segmentation task such as h-label
https://github.com/openvinotoolkit/training_extensions/blob/85a38dcd5a96b8d4a80c79d0df7f761c2565e1b7/src/otx/core/data/dataset/classification.py#L121
This class should warn users if there is no `background` labels in datumaro label categories and assign it to `label=0`.